File: cards_against_humanity/lib/models/player.dart
class Player {
  final String id;
  String nickname;
  int score = 0;
  bool isMyTurn = false;
  bool isReady = false;

  Player(this.id, this.nickname);
  Player.inLobby(this.id, this.nickname, this.isReady);
  Player.inGame(
      this.id, this.nickname, this.isReady, this.score, this.isMyTurn);
  void setNickname(String newNickname) {
    nickname = newNickname;
  }
}

File: cah.server/src/main/java/com/touchgrass/cah/server/model/Player.java
package com.touchgrass.cah.server.model;

import lombok.*;

import java.util.ArrayList;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Player {
    private String id;
    private String nickname;
    private boolean isReady = false;
    private boolean isMyTurn = false;
    private int score = 0;
    private ArrayList<WhiteCard> hand = new ArrayList<>();

    public Player(String id, String nickname) {
        this.id = id;
        this.nickname = nickname;
    }
}
